packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -51,21 +52,32 @@ class WebEntrypointTarget extends Target {
     final String targetFile = environment.defines[kTargetFile];
     final bool shouldInitializePlatform = environment.defines[kInitializePlatform] == 'true';
     final bool hasPlugins = environment.defines[kHasWebPlugins] == 'true';
-    final String import = fs.file(fs.path.absolute(targetFile)).uri.toString();
+    final String importPath = fs.path.absolute(targetFile);
+    final PackageUriMapper packageUriMapper = PackageUriMapper(
+      importPath,
+      PackageMap.globalPackagesPath,
+      null,
+      null,
+    );
+    final Uri mainImport = packageUriMapper.map(importPath);
+    if (mainImport == null) {
```

Comment:
technically, this is not a problem: we only have a problem when it's possible to import the same file via `package:*` imports.